client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx

```diff
@@ -0,0 +1,213 @@
+		removePurchase: PropTypes.func.isRequired,
+		closeDialog: PropTypes.func.isRequired,
+		purchase: PropTypes.object,
+	};
+
+	state = {
+		step: 1,
+		domainValidated: false,
+		userAgreed: false,
+		showErrors: false,
+	};
+
+	renderFirstStep( productName ) {
+		const { translate, purchase } = this.props;
+
+		return (
+			<Fragment>
+				<FormSectionHeading>
+					{ translate( '{{strong}}Delete{{/strong}} %(domain)s and remove it from your account.', {
+						args: { domain: productName },
+						components: { strong: <strong /> },
+					} ) }
+				</FormSectionHeading>
+				<p>
+					{ translate(
+						'Deleting {{strong}}%(domain)s{{/strong}} is {{strong}}permanent{{/strong}}. ' +
+							'You will no longer own it, and it could be registered ' +
+							'by someone else.',
+						{
+							args: { domain: productName },
+							components: { strong: <strong /> },
+						}
+					) }
+				</p>
+				{ ! isRefundable( purchase ) && maybeWithinRefundPeriod( purchase ) && (
+					<p>
+						<strong>
+							{ translate(
+								"We're not able to refund this purchase automatically. " +
+									"If you're canceling within %(refundPeriodInDays)s days of " +
+									'purchase, {{contactLink}}contact us{{/contactLink}} to ' +
+									'request a refund.',
+								{
+									args: {
+										refundPeriodInDays: purchase.refundPeriodInDays,
+									},
+									components: {
+										contactLink: <a href={ CALYPSO_CONTACT } />,
+									},
+								}
+							) }
+						</strong>
+					</p>
+				) }
+			</Fragment>
+		);
+	}
+
+	onDomainChange = event => {
+		const productName = getName( this.props.purchase );
+		this.setState( { domainValidated: event.currentTarget.value === productName } );
+	};
+
+	onAgreeChange = event => {
+		this.setState( { userAgreed: event.target.checked } );
+	};
+
+	renderSecondStep( productName ) {
+		const { translate } = this.props;
+
+		return (
+			<Fragment>
+				<FormSectionHeading>
+					{ translate(
+						'{{strong}}Delete{{/strong}} this domain by typing "%(domain)s" into the field below:',
+						{
+							args: { domain: productName },
+							components: { strong: <strong /> },
+						}
+					) }
+				</FormSectionHeading>
+				<FormFieldset>
+					<FormLabel>
+						{ translate( 'Type your domain name to proceed', { context: 'Domain name' } ) }
+						<FormTextInput
+							name="domain"
+							isError={ this.state.showErrors && ! this.state.domainValidated }
+							onChange={ this.onDomainChange }
+						/>
+						{ this.state.showErrors && ! this.state.domainValidated && (
+							<FormInputValidation
+								text={ translate( 'The domain name you entered does not match.' ) }
+								isError
+							/>
+						) }
+					</FormLabel>
+				</FormFieldset>
+				<FormFieldset>
+					<FormLabel>
+						<FormCheckbox name="agree" value="true" onChange={ this.onAgreeChange } />
+						<span>
+							{ translate( 'I understand that this is permanent and irreversible.', {
```

Comment:
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?